x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -35,9 +36,12 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class TransportEsqlQueryAction extends HandledTransportAction<EsqlQueryRequest, EsqlQueryResponse> {
 
+    public static TimeValue DEFAULT_TIMEOUT = new TimeValue(4, TimeUnit.MINUTES);
+    public static TimeValue MAX_TIMEOUT = new TimeValue(4, TimeUnit.MINUTES);
```

Comment:
The default and max values are supposed to be lower than other timeouts that can happen in the infrastructure, eg. the a http timeouts on cloud proxies.

Reply:
We shouldn't have MAX_TIMEOUT since ES can be deployed to an environment without the proxy timeout.

Reply:
Maybe we can move it to a Setting and disable it by default, so that we can enable it when we know we have a proxy timeout.